pyvrp/cpp/crossover/crossover.cpp

```diff
@@ -8,35 +9,35 @@ namespace
 {
 struct InsertPos  // best insert position, used to plan unplanned clients
 {
-    int deltaCost;
+    cost_type deltaCost;
     Route *route;
     size_t offset;
 };
 
 // Evaluates the cost change of inserting client between prev and next.
-int deltaCost(Client client, Client prev, Client next, ProblemData const &data)
+cost_type
+deltaCost(Client client, Client prev, Client next, ProblemData const &data)
 {
-    int prevEarliestArrival
+    duration_type const prevEarliestArrival
         = std::max(data.dist(0, prev), data.client(prev).twEarly);
-    int prevEarliestFinish
+    auto const prevEarliestFinish
         = prevEarliestArrival + data.client(prev).serviceDuration;
-    int distPrevClient = data.dist(prev, client);
-    int clientLate = data.client(client).twLate;
+    auto const clientLate = data.client(client).twLate;
 
-    if (prevEarliestFinish + distPrevClient >= clientLate)
-        return INT_MAX;
+    if (prevEarliestFinish + data.duration(prev, client) >= clientLate)
+        return cost_type(INT_MAX);
 
-    int clientEarliestArrival
+    duration_type const clientEarliestArrival
         = std::max(data.dist(0, client), data.client(client).twEarly);
-    int clientEarliestFinish
+    auto const clientEarliestFinish
         = clientEarliestArrival + data.client(client).serviceDuration;
-    int distClientNext = data.dist(client, next);
-    int nextLate = data.client(next).twLate;
+    auto const nextLate = data.client(next).twLate;
 
-    if (clientEarliestFinish + distClientNext >= nextLate)
-        return INT_MAX;
+    if (clientEarliestFinish + data.duration(client, next) >= nextLate)
+        return cost_type(INT_MAX);
```

Comment:
Wouldn't it be better to define a constant `COST_MAX` of the right type? Now it relies on doubles being able to represent correctly INT_MAX.

Reply:
I've replaced this with `std::numeric_limits<cost_type>::max()`.